client/my-sites/woocommerce/woop/landing-page.tsx

```diff
@@ -9,29 +10,64 @@ import Image04 from 'calypso/assets/images/woocommerce/woop-cta-image04.jpeg';
 import CtaSection from 'calypso/components/cta-section';
 import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
 import MasonryWave from 'calypso/components/masonry-wave';
+import { addQueryArgs } from 'calypso/lib/url';
 
 interface Props {
 	startSetup: () => void;
+	isFeatureActive: boolean;
+	upgradingPlan: Record< string, unknown >;
+	siteSlug: string;
 }
 
 const images = [ { src: Image01 }, { src: Image02 }, { src: Image03 }, { src: Image04 } ];
 
-const WoopLandingPage: React.FunctionComponent< Props > = ( { startSetup } ) => {
+const WoopLandingPage: React.FunctionComponent< Props > = ( {
+	startSetup,
+	siteSlug,
+	isFeatureActive,
+	upgradingPlan,
+} ) => {
+	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce', href: `/woocommerce-installation` } ];
 	const ctaRef = useRef( null );
 
+	function onCTAHandler() {
+		if ( ! isFeatureActive && upgradingPlan?.product_slug ) {
+			return ( window.location.href = addQueryArgs(
+				{ redirect_to: window.location.href },
+				`/checkout/${ siteSlug }/${ upgradingPlan.product_slug }`
+			) );
+		}
+
+		startSetup();
+	}
+
+	const upgradeButtonText = sprintf(
+		/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
+		__( 'Upgrade to %s' ),
+		upgradingPlan.product_name
+	);
+
+	const headlineText = isFeatureActive
+		? __( 'Build exactly the eCommerce website you want.' )
+		: sprintf(
+				/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
+				__( 'Upgrade to the %s plan and set up your WooCommerce store.' ),
```

Comment:
Similar to the comment above about the plan name. We could maybe not render anything until it's available?